Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -334,28 +338,21 @@ longest_border(const PolygonMesh& pmesh,
             typename property_map_value<PolygonMesh, CGAL::vertex_point_t>::type>::Kernel::FT  FT;
   typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor                       halfedge_descriptor;
 
-  std::unordered_set<halfedge_descriptor> visited;
+  std::deque<halfedge_descriptor> boundary_cycles;
+  extract_boundary_cycles(pmesh, std::back_inserter(boundary_cycles));
   halfedge_descriptor result_halfedge = boost::graph_traits<PolygonMesh>::null_halfedge();
   FT result_len = 0;
-  for(halfedge_descriptor h : halfedges(pmesh))
+  for(halfedge_descriptor h : boundary_cycles)
   {
-    if(visited.find(h)== visited.end())
+    FT len = 0;
+    for(halfedge_descriptor haf : halfedges_around_face(h, pmesh))
     {
-      if(is_border(h, pmesh))
-      {
-        FT len = 0;
-        for(halfedge_descriptor haf : halfedges_around_face(h, pmesh))
-        {
-          len += edge_length(haf, pmesh, np);
-          visited.insert(haf);
-        }
-
-        if(result_len < len)
-        {
-          result_len = len;
-          result_halfedge = h;
-        }
-      }
+      len += edge_length(haf, pmesh, np);
+    }
```

Comment:
Just call `face_border_length(haf, pmesh, np)`?

Reply:
done. thanks for pointing out.